atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/TableRendererV2.java

```diff
@@ -0,0 +1,422 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.table.description.render;
+
+import static com.palantir.atlasdb.table.description.render.ColumnRenderers.VarName;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.SortedSet;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import javax.annotation.Generated;
+import javax.lang.model.element.Modifier;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Supplier;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.table.description.NameComponentDescription;
+import com.palantir.atlasdb.table.description.NameMetadataDescription;
+import com.palantir.atlasdb.table.description.NamedColumnDescription;
+import com.palantir.atlasdb.table.description.Schemas;
+import com.palantir.atlasdb.table.description.TableDefinition;
+import com.palantir.atlasdb.table.description.TableMetadata;
+import com.palantir.atlasdb.table.generation.ColumnValues;
+import com.palantir.atlasdb.transaction.api.Transaction;
+import com.palantir.common.base.BatchingVisitableView;
+import com.palantir.common.persist.Persistables;
+import com.squareup.javapoet.AnnotationSpec;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+
+public class TableRendererV2 {
+    private final String packageName;
+    private final Namespace namespace;
+
+    public TableRendererV2(String packageName, Namespace namespace) {
+        Preconditions.checkNotNull(packageName);
+        Preconditions.checkNotNull(namespace);
+        this.packageName = packageName;
+        this.namespace = namespace;
+    }
+
+    public String getClassName(String rawTableName, TableDefinition table) {
+        return Renderers.getClassTableName(rawTableName, table) + "SimpleTable";
```

Comment:
since this `"SimpleTable"` string shows up in a few places, let's extract it to a constant somewhere